test/testutil/main.c

```diff
@@ -25,7 +25,7 @@ static void check_arg_usage(void)
 
     for (i = 0; i < n; i++)
         if (!arg_used[i+1])
-            test_printf_stderr("Warning ignored command-line argument %d: %s\n",
+            test_printf_stderr("Warning ignored command-line argument %zu: %s\n",
```

Comment:
Portability note, we already use '%z', so this should not cause any new issues.

Reply:
It's a trick question. We support %z with BIO_printf, but it's not given that we want to make assumption about its availability in arbitrary run-time. I mean since test_printf_stderr can end up in vprintf, which we have no control over. This means that it would be more appropriate to resolve this with explicit cast of the argument, it would be more portable.

Reply:
Does OpenSSL require any C standard? I.e. what is the minimum C standard required? C89? What do we cast to? Long? Because to do it "right" in old MSVC I think we need "I64"?

Reply:
I'd say that it's rather a de-facto thing than firm requirement. At least as long as there are arcane platforms on the list you are kind of driven to being reluctant to going "all-in" with overly specific requirement and tend to opting for least possible assumptions. Yet you don't want to preclude possibility of using novel features, which effectively means #if-ing stuff. But at the same time you want to minimize #if-ing, right? So it becomes compromise matter, i.e. do you gain enough from #if-ing? For example, do you gain enough from #if-ing atomics? Yes. Do you gain enough from #if-ing "I64" here? I'd say no, it doesn't worth it. The value in question is safe to cast to 32 bits even on 64-bit platform. Because it's capped at 1000. So I'd say it's appropriate to resolve the trade-off with cast to int or unsigned int. [This naturally applies to next test_printf_stderr in subroutine in question.]

Reply:
Yes, the #if-ing is a losing proposition, because you keep forgetting to do the special handling...
The question is: Is anyone still using arcane platforms, or they have all died and we can move on to c89 at least?

Reply:
It's easy to pop the question, I wish it was as easy to answer. And related question is if answer would actually give you peace of mind. Let's say a problem with HP-UX is reported, what do I do? Latest HP-UX is likely to support %z, but one I have access to doesn't. But I can pass -std=c99... Life is not black-n-white... And I'm kind of tired of making the point that it's not... Is it actually unreasonable suggestion to make least possible assumption?

Reply:
I suggest that you (openssl maintainers) do the following: Let's put in the next release notes that we are planning not to support anything older than c89 or c99 and if people still need this support to contact you. Perhaps nobody is using ancient stuff anymore and we can move forward to more reasonable standards. For now a cast will do.

Reply:
> I suggest that you

Oh! We couldn't possibly think of it! ... Come on, man, this is insulting my intelligence :-) :-) :-) But on serious note, the suggestion is effectively as flawed as current state because of "contact" clause. I mean if somebody contacts with corresponding request what would be the response? "Allow us to sanitize all dependencies on latest C standard"? Of course not. The only possible answer would be "go away", naturally politely formulated. And if this is the only possible answer, then why bother? It would be way more honest to just remove arcane targets, and if anybody complains, they should be advised to remain with older versions. That was my suggestion (for 1.1.1). At the same time I for one would still be reluctant to be as categoric as "C99 all the way or go away". As already said, life is not black-n-white and acknowledging it doesn't have to cost a lot of effort. This %z thing being perfect example. BTW, requirement for C90 is verbalized in coding style guide.

Reply:
If we state: Starting with 1.2.0 we are going to start using C99 features in the code. If your system does not support such a compiler, please let us know. I suspect that nobody will respond. Then we don't just go and convert all the code, but we start to use new features.

Reply:
C99 was actually brought up recently in team discussions...  and then someone reminded us that MSVC doesn't support all C99 features, only enough to compile comfortably as C++.  A look around on the net tells me that this will never change, Microsoft seems to have stopped developing their C compiler.

So if we're going to switch up, we need to explore what that means in MSVC terms first.

It was also mentioned that there's no *actual need* to switch up, at least from a language syntax point of view (and our decision for C89/C90 as minimum requirement is pure a supported language syntax thing)...  and that's true as well, a switch up would be a coolness factor more than anything else (I mean, yeah sure, you can have variable declarations anywhere in the code and you can have `//` comment starters), but *shrug*.

Also, we must remember that OpenSSL is used in very oddball environments, possibly ones that have yet to see the shadow of C99...

Reply:
Formally speaking this is like worst place to discuss this, so I suggest to concentrate on request's matter [and take the standard compliance dispute elsewhere if so desired].